### PR TITLE
Tracing Spans of RPCs should follow otel semantic conventions

### PR DESCRIPTION
**Pull request by ndimiduk**
A first set of patches to bring our implementation around to the conventions as described. Focuses primarily on spans from HTable and a couple other connection-level methods. Applicable reference is https://github.com/open-telemetry/opentelemetry-specification/blob/3e380e2/specification/trace/semantic_conventions/database.md

### CODE REVIEW

File: hbase-client/src/test/java/org/apache/hadoop/hbase/client/TestAsyncRegionLocatorTracing.java [hunk 1]
```diff
@@ -90,16 +102,33 @@ public void tearDown() throws IOException {
...
   }
 
   @Test
   public void testClearCache() {
     conn.getLocator().clearCache();
     SpanData span = waitSpan("AsyncRegionLocator.clearCache");
-    assertEquals(StatusCode.OK, span.getStatus().getStatusCode());
+    assertThat(span, allOf(
+      hasStatusWithCode(StatusCode.OK),
+      hasKind(SpanKind.CLIENT),
```
Comment (ndimiduk) [inline hbase-client/src/test/java/org/apache/hadoop/hbase/client/TestAsyncRegionLocatorTracing.java:130]: I'm not clear on whether these spans should be "client" or "internal". Perhaps only the ones that can make an RPC should be "client", the rest left as "internal".

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  0s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 26s |  master passed  |
| +1 :green_heart: |  compile  |   5m  3s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m  6s |  master passed  |
| +1 :green_heart: |  spotbugs  |   4m  4s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 15s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 47s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  javac  |   0m 55s |  hbase-client generated 0 new + 121 unchanged - 1 fixed = 121 total (was 122)  |
| +1 :green_heart: |  javac  |   3m 19s |  hbase-server in the patch passed.  |
| -0 :warning: |  checkstyle  |   0m 27s |  hbase-client: The patch generated 12 new + 1 unchanged - 0 fixed = 13 total (was 1)  |
| -0 :warning: |  checkstyle  |   1m 13s |  hbase-server: The patch generated 5 new + 147 unchanged - 0 fixed = 152 total (was 147)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  21m 29s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotbugs  |   4m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 33s |  The patch does not generate ASF License warnings.  |
|  |   |  64m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3863 |
| Optional Tests | dupname asflicense javac hadoopcheck xml compile spotbugs hbaseanti checkstyle |
| uname | Linux f170a6f25865 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b2571df7ae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 42s |  master passed  |
| +1 :green_heart: |  compile  |   2m  9s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 19s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 34s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  8s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  8s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 14s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  5s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 23s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 139m  1s |  hbase-server in the patch failed.  |
|  |   | 179m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3863 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 4fc3f8de33aa 4.15.0-156-generic #163-Ubuntu SMP Thu Aug 19 23:31:58 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b2571df7ae |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/testReport/ |
| Max. process+thread count | 4416 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 50s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 32s |  master passed  |
| +1 :green_heart: |  compile  |   1m 54s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 24s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 30s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 56s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 33s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 159m 35s |  hbase-server in the patch failed.  |
|  |   | 201m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3863 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 1332c1fbdd49 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b2571df7ae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/testReport/ |
| Max. process+thread count | 5296 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3863/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Let's open separated PRs for the different issues? It will be easier to review

**Comment by Apache9** [comment]
Hey @ndimiduk , any change of splitting this into several PRs? We can merge them one by one.

Thanks.

**Comment by ndimiduk** [comment]
@Apache9 sure, if you prefer, we can do them one-by-one. I thought having the whole change-set on a feature branch / single PR would make it easier for reviewers to evaluate the changes as a whole.

**Comment by Apache9** [comment]
> @Apache9 sure, if you prefer, we can do them one-by-one. I thought having the whole change-set on a feature branch / single PR would make it easier for reviewers to evaluate the changes as a whole.

For me, I think the difficulty for the big PR is that, once I point out a problem, it will be a pain to find out which commit should be fixed, and the review process will spend more time, so conflict with the base branch will also be a pain...

**Pull request closed without merging**